tests/Task/OnCompletion.test.ts

```diff
@@ -0,0 +1,197 @@
+            .dueDate(dueDate)
+            .build();
+        expect(task.status.type).toEqual(StatusType.TODO);
+
+        // Act
+        const returnedTasks = applyStatusAndOnCompletionAction(task, Status.makeDone());
+
+        // Assert
+        expect(returnedTasks.length).toEqual(1);
+        expect(toLines(returnedTasks).join('\n')).toMatchInlineSnapshot(
+            '"- [x] A non-recurring task with invalid OC trigger 🏁 INVALID_ACTION 📅 2024-02-10 ✅ 2024-02-11"',
+        );
+    });
+
+    it('should just return task if StatusType has not changed', () => {
+        // Arrange
+        const dueDate = '2024-02-10';
+        const task = new TaskBuilder()
+            .description('An already-DONE, non-recurring task 🏁 Delete')
+            .dueDate(dueDate)
+            .doneDate(dueDate)
+            .status(Status.DONE)
+            .build();
+        expect(task.status.type).toEqual(StatusType.DONE);
+
+        // Act
+        const returnedTasks = applyStatusAndOnCompletionAction(task, Status.makeDone());
+
+        // Assert
+        expect(returnedTasks.length).toEqual(1);
+        expect(toLines(returnedTasks).join('\n')).toMatchInlineSnapshot(
+            '"- [x] An already-DONE, non-recurring task 🏁 Delete 📅 2024-02-10 ✅ 2024-02-10"',
+        );
+    });
+
+    it('should just return trigger-less, non-recurring task', () => {
+        // Arrange
+        const dueDate = '2024-02-10';
+        const task = new TaskBuilder().description('A non-recurring task with no trigger').dueDate(dueDate).build();
+        expect(task.status.type).toEqual(StatusType.TODO);
+
+        // Act
+        const tasks = applyStatusAndOnCompletionAction(task, Status.makeDone());
+
+        // Assert
+        expect(tasks.length).toEqual(1);
+        expect(toLines(tasks).join('\n')).toMatchInlineSnapshot(
+            '"- [x] A non-recurring task with no trigger 📅 2024-02-10 ✅ 2024-02-11"',
+        );
+    });
+
+    it('should just return trigger-less recurring task', () => {
+        // Arrange
+        const dueDate = '2024-02-10';
+        const recurrence = new RecurrenceBuilder().rule('every day').dueDate(dueDate).build();
+        const task = new TaskBuilder()
+            .description('A recurring task with no trigger')
+            .recurrence(recurrence)
+            .dueDate(dueDate)
+            .build();
+        expect(task.status.type).toEqual(StatusType.TODO);
+
+        // Act
+        const tasks = applyStatusAndOnCompletionAction(task, Status.makeDone());
+
+        // Assert
+        expect(tasks.length).toEqual(2);
+        expect(toLines(tasks).join('\n')).toMatchInlineSnapshot(`
+            "- [ ] A recurring task with no trigger 🔁 every day 📅 2024-02-11
+            - [x] A recurring task with no trigger 🔁 every day 📅 2024-02-10 ✅ 2024-02-11"
+        `);
+    });
+
+    // TODO is there a better way to handle the following?  does an 'empty' Task exist?
```

Comment:
There is no better way. An empty array is the correct answer in this case, given that a `Task[]` is being returned.

Returning anything other than an empty array would require callers to write special-case code.